### PR TITLE
second prototype: open sidebar with information

### PR DESCRIPTION
**Pull request by Yifan-Zeng**
Second prototype: the sidebar open with information in it
the design looks like the following picture:
Capture d’écran 2021-06-04 à 09 49 57

### CODE REVIEW

File: app/src/app/poll/poll.component.html [hunk 1]
```diff
@@ -86,90 +86,99 @@ <h4 *ngIf="displayedResult !== undefined" i18n>Results of round {{ displayedResu
...
-            <ng-container *ngSwitchCase="true">
-              your previous vote will be carried forward.
-            </ng-container>
-          </ng-container>
-          The round will end {{ answer.RoundDeadline | nearDate:'inside' }} or when all participants have voted,
-          whichever comes first.
-        </p>
+  <div class="infomenu">
+    <table>
+      <tr>
```
Comment (Jogo27) [inline app/src/app/poll/poll.component.html:95]: No. You're not displaying a table, hence you can not use a table element. Using table to ease your layout is just plain wrong. Instead, you could use modern CSS layout possibilities like [flex](https://www.w3.org/TR/css-flexbox-1/) and [grid](https://www.w3.org/TR/css-grid-1/).

File: app/src/app/poll/poll.component.html [hunk 2]
```diff
@@ -86,90 +86,99 @@ <h4 *ngIf="displayedResult !== undefined" i18n>Results of round {{ displayedResu
...
-        <ng-container *ngIf="!hasCurrentRoundBallot()">
-          It will not be possible to join the poll once the first round has ended.
-        </ng-container>
-        Then, each round will last {{ answer.MaxRoundDuration | duration }}, or until all participants have voted,
-        whichever comes first.
-      </p>
-    </ng-template>
+  <div class="simplemenu">
+    <mat-icon class="helpbutton" id="helpbutton">help</mat-icon>
+  </div>
```
Comment (Jogo27) [inline app/src/app/poll/poll.component.html:91]: Why is this needed ?

File: app/src/styles.sass [hunk 3]
```diff
@@ -142,18 +138,28 @@ $itero-mat-theme: mat-light-theme(( color: ( primary: $itero-mat-primary, accent
...
-  box-shadow: -2px 0px 2px 0 rgba(255, 204, 244, 0.9)
+.infomenu
+  width: 200px
+  margin-right: 0
+  position: absolute
+  right: 0
   .helpbutton
-    text-align: center
+    position: relative
+    top: -280px
```
Comment (Jogo27) [inline app/src/styles.sass:162]: Try to avoid using such absolute values in pixels. You could instead change the vertical alignment for the icon to be on top.

File: app/src/styles.sass [hunk 4]
```diff
@@ -142,18 +138,28 @@ $itero-mat-theme: mat-light-theme(( color: ( primary: $itero-mat-primary, accent
...
+  box-shadow: -5px 0px 5px 0 rgba(255, 204, 244, 0.7)
+    
+.simplemenu
+  display: none
   height: 100%
   margin-right: -16px
-  margin-left: 3px
-  box-shadow: -2px 0px 2px 0 rgba(255, 204, 244, 0.9)
+.infomenu
+  width: 200px
```
Comment (Jogo27) [inline app/src/styles.sass:156]: Try to avoid using such absolute value in pixels. Here you could use percents. Moreover, I think it could be made wider.

**Review by Jogo27 — CHANGES_REQUESTED** [review]
Nice! There are a few things to fix though. In addition to list below look at these two screenshots.
image
image

**Comment by Yifan-Zeng** [comment]
PTAL 
(I don't know the height of the bar should be like what so I didn't set the height, but
when testing it, you can add "height: 80%" for ".infomenu" to have a better look effect, thank you)

File: app/src/app/app.component.sass [hunk 5]
```diff
@@ -16,6 +16,7 @@ footer
       img:not(#irit-logo)
         position: relative
         top: -4px
+        z-index: -1
```
Comment (Jogo27) [inline app/src/app/app.component.sass:19]: No. z-index must reflect how the boxes of the page are displayed in the axis perpendicular to the screen. Here we do not want some icons of the footer to stay behind the main part of the page (which always has z-index: 0). We want the information bar to be in front of the main part of the page.

File: app/src/app/poll/poll.component.html [hunk 6]
```diff
@@ -86,90 +86,93 @@ <h4 *ngIf="displayedResult !== undefined" i18n>Results of round {{ displayedResu
...
-      <ng-container *ngSwitchCase="'Waiting'">
-        <p i18n>
-          The poll will start {{ answer.Start | nearDate:'inside' }}.
-          You won't be able to vote until that date.
-        </p>
-      </ng-container>
-      <ng-container *ngSwitchCase="'Terminated'">
+  <div class="infomenu">
+    <div class="infomenuincon">
+      <mat-icon class="helpbutton" id="helpbutton">help</mat-icon>
```
Comment (Jogo27) [inline app/src/app/poll/poll.component.html:93]: CSS names are much easier to read when words are separated with `-`. I regret to not have done it since the beginning for the class and id names. Moreover, the information "bar" is not a menu. Hence I suggest info-panel instead of infomenu, info-panel-icon instead of infomenuincon, and help-button instead of helpbutton.

Comment (Jogo27) [inline app/src/app/poll/poll.component.html:93]: There is no reason here to use the same name for the class and id of the same element. Use it for the id only.

File: app/src/styles.sass [hunk 7]
```diff
@@ -142,18 +138,25 @@ $itero-mat-theme: mat-light-theme(( color: ( primary: $itero-mat-primary, accent
 @include angular-material-typography($itero-mat-typography)
 
 
-.sidemenu
+
+.infomenu
   border-left:1px solid #704f6a
   border-top: 1px solid #704f6a
   border-bottom: 1px solid #704f6a
```
Comment (Jogo27) [inline app/src/styles.sass:145]: Why not
```suggestion
  border: 1px solid #704f6a
  border-right: none
```

File: app/src/styles.sass [hunk 8]
```diff
@@ -142,18 +138,25 @@ $itero-mat-theme: mat-light-theme(( color: ( primary: $itero-mat-primary, accent
...
   background-color: #d5b6d0
   float: right
```
Comment (Jogo27) [inline app/src/styles.sass:147]: Seems useless to me.

File: app/src/styles.sass [hunk 9]
```diff
@@ -142,18 +138,25 @@ $itero-mat-theme: mat-light-theme(( color: ( primary: $itero-mat-primary, accent
...
-  height: 100%
-  margin-right: -16px
   margin-left: 3px
```
Comment (Jogo27) [inline app/src/styles.sass:148]: Seems useless to me.

File: app/src/styles.sass [hunk 10]
```diff
@@ -142,18 +138,25 @@ $itero-mat-theme: mat-light-theme(( color: ( primary: $itero-mat-primary, accent
...
-  box-shadow: -2px 0px 2px 0 rgba(255, 204, 244, 0.9)
-  .helpbutton
-    text-align: center
+  box-shadow: -5px 0px 5px 0 rgba(255, 204, 244, 0.7)
+  width: 80%
+  margin-right: 0
```
Comment (Jogo27) [inline app/src/styles.sass:151]: Seems useless to me.

File: app/src/styles.sass [hunk 11]
```diff
@@ -142,18 +138,25 @@ $itero-mat-theme: mat-light-theme(( color: ( primary: $itero-mat-primary, accent
...
+  position: absolute
+  right: 0
+  display: flex
+  flex-flow: row nowrap
+  .infomenuicon
+    display: flex 
+    .helpbutton
+      align-self: flex-start
```
Comment (Jogo27) [inline app/src/styles.sass:159]: These 4 lines are not used because there is a typo in the class name in the HTML file (infomenui**n**con). This means that you can remove them.

**Review by Jogo27 — CHANGES_REQUESTED** [review]
It seems to me that it is much better to **not** add `height: 80%` and have the box adapted to the text it contains.

The PR looks almost done now. Apart from the minor points below, please just add some space around the icon.

**Comment by Yifan-Zeng** [comment]
PTAL

File: app/src/app/poll/poll.component.html [hunk 12]
```diff
@@ -86,90 +86,93 @@ <h4 *ngIf="displayedResult !== undefined" i18n>Results of round {{ displayedResu
...
-          The poll will start {{ answer.Start | nearDate:'inside' }}.
-          You won't be able to vote until that date.
-        </p>
-      </ng-container>
-      <ng-container *ngSwitchCase="'Terminated'">
+  <div class="info-panel">
+    <div class="info-panel-icon">
+      <mat-icon id="help-button">help</mat-icon>
+    </div>
+    <div class="info-panel-info" *ngIf="!!answer">
```
Comment (Jogo27) [inline app/src/app/poll/poll.component.html:95]: `info-panel-content` ?

File: app/src/styles.sass [hunk 13]
```diff
@@ -142,18 +138,24 @@ $itero-mat-theme: mat-light-theme(( color: ( primary: $itero-mat-primary, accent
...
+  box-shadow: -5px 0px 5px 0 rgba(255, 204, 244, 0.7)
+  width: 80%
+  margin-right: 0
+  position: absolute
+  right: 0
+  display: flex
+  flex-flow: row nowrap
+  z-index: 1
+  .info-panel-icon 
+    #help-button
```
Comment (Jogo27) [inline app/src/styles.sass:157]: There is no need for two selectors here. There is nothing else in the first rule than another rule. Use one selector only, with a [descendant combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/Descendant_combinator).

File: app/src/app/poll/poll.component.html (hunk 12 above)
Comment (Jogo27) [inline app/src/app/poll/poll.component.html:95]: Class `information` had typographic properties attached to it in styles.sass. Removing this class removed this typographic properties. Perhaps you could have two classes for this element.

File: app/src/styles.sass [hunk 14]
```diff
@@ -142,18 +138,24 @@ $itero-mat-theme: mat-light-theme(( color: ( primary: $itero-mat-primary, accent
...
+      margin: 17px 3px
```
Comment (Jogo27) [inline app/src/styles.sass:158]: I understand that you want this margin to be the same as the margin of the p element in div.info-panel-info. But you cannot rely on the fact that this margin is approximatively 17px, because
1. the margin of the p element is expressed in ex (or em) unit, which means that it depend on the font used;
2. this is a default property of p, meaning that in may depend on the browser.

For the horizontal margin, maybe could you use a variable from size.sass?

**Review by Jogo27 — CHANGES_REQUESTED** [review]
I'm sorry but you introduced new issues and you haven't resolved all the previous one. If you don't agree with some comments, please answer them.

**Comment by Yifan-Zeng** [comment]
PTAL

**Comment by Yifan-Zeng** [comment]
PTAL

File: app/src/styles.sass [hunk 15]
```diff
@@ -5,6 +5,9 @@
 @import 'mixins'
 @import 'responsive'
 
+$breakpoint-tablet: 500px
+$breakpoint-laptop: 1000px
+
```
Comment (Jogo27) [inline app/src/styles.sass:10]: No. These variables must be defined in responsive.sass (only).

File: app/src/styles.sass [hunk 16]
```diff
@@ -152,18 +153,27 @@ $itero-mat-theme: mat-light-theme(( color: ( primary: $itero-mat-primary, accent
...
+  box-shadow: -5px 0px 5px 0 rgba(255, 204, 244, 0.7)
+  width: 80%
+  position: absolute
+  right: 0
+  display: flex
+  flex-flow: row nowrap
+  @media (min-width: $breakpoint-tablet)
+    width: 50%
+  @media (min-width: $breakpoint-laptop)
+    width: 30%
```
Comment (Jogo27) [inline app/src/styles.sass:170]: On screens with width greater than 1000px, we won't use the information panel. Hence these lines are of little use and will introduce useless complications. Please remove them.

File: app/src/styles.sass [hunk 17]
```diff
@@ -152,18 +153,27 @@ $itero-mat-theme: mat-light-theme(( color: ( primary: $itero-mat-primary, accent
...
+  .info-panel-icon #help-button
+    margin: 0.7em $small-space
+  .info-panel-content .information
+    margin-left: 0
+    border-left: 0
+    padding-left: 0
```
Comment (Jogo27) [inline app/src/styles.sass:176]: You could use `display: contents` instead of these three lines. Another solution is to not have a div.information inside a div.info-panel-content, but just one div with the two classes `<div class="info-panel-content information">`.

**Review by Jogo27 — CHANGES_REQUESTED** [review]
This is happening again:
image

Moreover, for CreateComponent to be displayed as before, please **do not** reduce the maximum width of the second column of .information-container to 5% when the screen is wide enough.

**Comment by Yifan-Zeng** [comment]
PTAL
For poll page: for the mobile screen, sidebar of information is 80%; for pads it takes 50%; for laptop the sidebar disappears for the moment(replace by the information on right like origin later on). 
For create page, for laptops, origin design (31% for the information on side); for mobiles and pads reduced design (5% for the information, we will change it later on to apply our design of sidebar)

File: app/src/styles.sass [hunk 18]
```diff
@@ -5,6 +5,7 @@
 @import 'mixins'
 @import 'responsive'
 
+
```
Comment (Jogo27) [inline app/src/styles.sass:8]: Please remove this useless empty line.

File: app/src/styles.sass [hunk 19]
```diff
@@ -131,15 +132,16 @@ input
 .information-container
   display: grid
   grid-template-columns: 1fr minmax(min-content, 5%)
+  @media screen and (min-width: $breakpoint-laptop)
+    grid-template-columns: 1fr minmax(min-content, 31%)
+
```
Comment (Jogo27) [inline app/src/styles.sass:137]: Please remove this useless empty line.

File: app/src/styles.sass [hunk 20]
```diff
@@ -131,15 +132,16 @@ input
 .information-container
   display: grid
   grid-template-columns: 1fr minmax(min-content, 5%)
+  @media screen and (min-width: $breakpoint-laptop)
+    grid-template-columns: 1fr minmax(min-content, 31%)
```
Comment (Jogo27) [inline app/src/styles.sass:136]: It should be in the other direction: by default the max-width of the second column is 31%, and on narrow screen it is shrunken to 5%.

File: app/src/styles.sass [hunk 21]
```diff
@@ -152,18 +154,26 @@ $itero-mat-theme: mat-light-theme(( color: ( primary: $itero-mat-primary, accent
...
   background-color: #d5b6d0
-  float: right
-  height: 100%
-  margin-right: -16px
-  margin-left: 3px
-  box-shadow: -2px 0px 2px 0 rgba(255, 204, 244, 0.9)
-  .helpbutton
-    text-align: center
+  box-shadow: -5px 0px 5px 0 rgba(255, 204, 244, 0.7)
+  width: 80%
```
Comment (Jogo27) [inline app/src/styles.sass:163]: Please move that line to line 168, such that it is closer to its modification when the screen is wider.

File: app/src/styles.sass [hunk 22]
```diff
@@ -131,15 +132,16 @@ input
...
+
 
 .information
   margin-left: $medium-space
   border-left: solid 1px $maincolor
   padding-left: $medium-space
-  display: none
   p
     font: $infofont
-    margin-top: 0
```
Comment (Jogo27) [inline app/src/styles.sass:142]: I need the margin-top to be 0, at least when the screen is wide.

**Review by Jogo27 — CHANGES_REQUESTED** [review]

File: app/src/styles.sass (hunk 21 above)
Comment (Yifan-Zeng) [reply app/src/styles.sass:163]: But when I delete line 163-168, it will be look like this 
Capture d’écran 2021-06-10 à 10 56 51

Comment (Jogo27) [reply app/src/styles.sass:163]: Yes. Please do **not** remove any of these lines. Just move line 163 such that is is just before the line with @media.

Comment (Yifan-Zeng) [reply app/src/styles.sass:163]: Ah okay I get it, just to change the position.

**Comment by Yifan-Zeng** [comment]
PTAL

**Review by Jogo27 — APPROVED** [review]
LGTM
